code/src/App.js

```diff
@@ -1,9 +1,98 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+
+import Header from 'components/Header';
+import NewPost from 'components/NewPost';
+import PostList from 'components/PostList';
 
 export const App = () => {
+  const [postList, setPostList] = useState([]);
+  const [newPost, setNewPost] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  const fetchData = () => {
+    setLoading(true);
+    fetch('https://happy-thoughts-technigo.herokuapp.com/thoughts')
+      .then((data) => data.json())
+      .then((transformedData) => setPostList(transformedData))
+      .catch((error) => console.error(error))
+      .finally(() => setLoading(false));
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  const onNewPostChange = (event) => {
+    setNewPost(event.target.value);
+  };
+
+  const handleFormCleanup = () => {
+    setNewPost('');
+    setLoading(false);
+  };
+
+  const handleFormSubmit = (event) => {
+    event.preventDefault();
+    const options = {
+      method: 'post',
+      body: JSON.stringify({
+        message: newPost
+      }),
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    };
+    setLoading(true);
+    fetch('https://happy-thoughts-technigo.herokuapp.com/thoughts', options)
+      .then((data) => data.json())
+      .then(() => fetchData())
+      .catch((error) => console.error(error))
+      .finally(() => handleFormCleanup());
+  };
+
+  if (loading) {
+    return (
+      <p>THE PAGE IS LOADING...</p>
+    )
+  }
+
+  const onLikesIncrease = (thoughtId) => {
+    const options = {
+      method: 'post',
+      headers: {
+        'content-type': 'application/json'
+      }
+    };
+
+    fetch(`https://happy-thoughts-technigo.herokuapp.com/thoughts/${thoughtId}/like`, options)
+      .then((res) => res.json())
+      .then((data) => {
+        const updatedThoughts = postList.map((item) => {
+          // eslint-disable-next-line no-underscore-dangle
```

Comment:
interesting that you put that comment here and not at the top, but it makes more sense here, the reader understands better why the  ESlint is disabled!